lib/deflation.cpp

```diff
@@ -101,24 +105,27 @@ namespace quda {
     const int nevs_to_print = param.cur_dim;
     if(nevs_to_print == 0) errorQuda("\nIncorrect size of current deflation space. \n"); 
 
-    Complex *projm  = new Complex [param.ld*param.cur_dim];
+    std::unique_ptr<Complex[] > projm(new Complex[param.ld*param.cur_dim]);
 
+    if (param.eig_global.extlib_type == QUDA_MAGMA_EXTLIB) {
 #ifdef MAGMA_LIB
-    memcpy(projm, param.matProj, param.ld*param.cur_dim*sizeof(Complex));
-    double *evals = new double[param.ld];
-
-    cudaHostRegister(static_cast<void *>(projm), param.ld*param.cur_dim*sizeof(Complex),  cudaHostRegisterDefault);
-    magma_Xheev(projm, param.cur_dim, param.ld, evals, sizeof(Complex));
-    cudaHostUnregister(projm);
+      memcpy(projm.get(), param.matProj, param.ld*param.cur_dim*sizeof(Complex));
+      std::unique_ptr<double[] > evals(new double[param.ld]);
 
-    delete [] evals;
+      cudaHostRegister(static_cast<void *>(projm.get()), param.ld*param.cur_dim*sizeof(Complex),  cudaHostRegisterDefault);
```

Comment:
Is there any reason not to use QUDA's `pinned_malloc` and `pinned_free` routines here? Or perhaps even better to use `pool_pinned_malloc` and `pool_pinned_free`?